x-pack/plugins/cloud_security_posture/public/components/fleet_extensions/setup_technology_selector/use_setup_technology.ts

```diff
@@ -38,8 +38,15 @@ export const useSetupTechnology = ({
     return SetupTechnology.AGENT_BASED;
   });
 
+  const [isDirty, setIsDirty] = useState<boolean>(false);
+
+  const updateSetupTechnology = (value: SetupTechnology) => {
+    setSetupTechnology(value);
+    setIsDirty(true);
+  };
+
   useEffect(() => {
```

Comment:
Hey Sean, thank you. I agree that using the `useEffect` should be avoided generally. I apply a fix because I believe this useEffect was placed before  to check edge case. cc: @maxcold  @kfirpeled  @seanrathier  For this bug useEffect is causing race condition for bugs. Do want to open a tech debt ticket or refactor in this PR? Refactoring can cause changes in architecture on how Agentless is setup and may take additional effort keep in mind.